src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -634,7 +634,14 @@ private async ValueTask WriteAsyncChunked<TIOAdapter>(TIOAdapter writeAdapter, R
         private ValueTask WriteSingleChunk<TIOAdapter>(TIOAdapter writeAdapter, ReadOnlyMemory<byte> buffer)
             where TIOAdapter : struct, IReadWriteAdapter
         {
-            byte[] rentedBuffer = ArrayPool<byte>.Shared.Rent(buffer.Length + FrameOverhead);
+            // For rented buffer size:
+            //     Prior to handshake; when _context is not yet set, we will use FrameOverhead.
```

Comment:
This seems like it's still not entirely correct, because if we're in the middle of a handshake, then the resulting header and trailer sizes could still be larger than FrameOverhead. Right?

Reply:
Yes; increased `FrameOverhead` to 64 and remarked its an estimation